elf/arch-riscv.cc

```diff
@@ -812,6 +841,43 @@ static void shrink_section(Context<E> &ctx, InputSection<E> &isec, bool use_rvc)
         delta += 4;
       break;
     }
+    case R_RISCV_HI20: {
+      // This relocation refers to an LUI instruction containing the high
+      // 20-bits to be relocated to an absolute symbol address.
+      // In some cases, LUI can be removed or replaced by C.LUI.
+
+      // If it's a static linking, we skip this relaxation for now, as it's
+      // currently not working.
+      if (ctx.arg.is_static)
+        break;
```

Comment:
I still don't understand why it fails in static linking. Looks like we accidentally removed some LUI instructions that shouldn't have been removed.